sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala

```diff
@@ -79,8 +79,16 @@ object FileSourceStrategy extends Strategy with Logging {
         ExpressionSet(normalizedFilters.filter(_.references.subsetOf(partitionSet)))
       logInfo(s"Pruning directories with: ${partitionKeyFilters.mkString(",")}")
 
+      val prunedFsRelation = fsRelation.location match {
```

Comment:
Can we push this pruning into the scan (i.e. do it when computing `inputRDD` in `FileSourceScanExec`)?